plugins/social_network/social_network.cpp

```diff
@@ -89,6 +171,213 @@ namespace golos { namespace plugins { namespace social_network {
         helper->select_active_votes(result, total_count, author, permlink, limit);
     }
 
+    void social_network::impl::set_depth_parameters(const content_depth_params &params) {
+        depth_parameters = params;
+    }
+
+    struct operation_visitor {
+        using result_type = void;
+
+        golos::chain::database &db;
+        content_depth_params depth_parameters;
+
+        operation_visitor(golos::chain::database &db, const content_depth_params &params) : db(db), depth_parameters(params) {
+        }
+
+        std::wstring utf8_to_wstring(const std::string &str) const {
+            return utf_to_utf<wchar_t>(str.c_str(), str.c_str() + str.size());
+        }
+
+        std::string wstring_to_utf8(const std::wstring &str) const {
+            return utf_to_utf<char>(str.c_str(), str.c_str() + str.size());
+        }
+
+        const comment_content_object &get_comment_content(const comment_id_type &comment) const {
+            try {
+                return db.get<comment_content_object, by_comment>(comment);
+            } FC_CAPTURE_AND_RETHROW((comment))
+        }
+
+        const comment_content_object *find_comment_content(const comment_id_type &comment) const {
+            return db.find<comment_content_object, by_comment>(comment);
+        }
+
+        template<class T>
+        void operator()(const T& o) const {
+        }
+
+        void operator()(const delete_comment_operation& o) const {
+            const auto &comment = db.get_comment(o.author, o.permlink);
+            auto& content = get_comment_content(comment.id);
+            db.remove(content);
+        }
+
+        void operator()(const golos::protocol::comment_operation& o) const {
+            if (depth_parameters.miss_content()) {
+                return;
+            }
+
+            const auto& comment = db.get_comment(o.author, o.permlink);
+
+            if (comment.created != db.head_block_time()) {
+                // Edit case
+                db.modify(db.get< comment_content_object, by_comment >( comment.id ), [&]( comment_content_object& con ) {
```

Comment:
The content can absent, so you should use `db.find<>` and check result on nullptr